README.md

```diff
@@ -1,49 +1,210 @@
-# Building
 
-Go to the directory with the tool and type in `cargo build` .
+![umi-transfer](docs/img/logoheader.svg)
 
-# Running
+# umi-transfer
 
-### Usage
+<p>
+    <b>A command line tool for transferring Unique Molecular Identifiers (UMIs) provided as separate FastQ file to the header of records in paired FastQ files.</b>
+</p>
 
-The tool requires an input as follows:
+<hr>
 
-> `umi-transfer [OPTIONS] <SUBCOMMAND> `<br>
+- [Background on Unique Molecular Identifiers](#background)
+- [Installing `umi-transfer`](#installation)
+- [Using `umi-transfer` to integrate UMIs](#usage)
+- [Improving performance with external multi-threaded compression](#high-performance-guide)
+- [Contributing bugfixes and new features](#contribution-guide-for-developers)
 
-`OPTIONS:`
-| Flag | Required | Description |
-| ------------- | :-----------: | ----------: |
-| `-h`,`--help` | No | Print help information |
-| `--prefix` | No, but default will be '`integrated`' | dictates name of output files|
-| `--r1-in` | Yes | FASTQ file with reads|
-| `--r2-in` | No | FASTQ file with reads |
+<hr>
 
-`SUBCOMMANDS: `
+## Background
 
-> `inline:`
->
-> > | Flag         |          Required          |                Description |
-> > | ------------ | :------------------------: | -------------------------: |
-> > | `--pattern1` |            Yes             | Nucleotide Pattern for UMI |
-> > | `--pattern2` | Needed if `--r2-in` exists | Nucleotide Pattern for UMI |
->
-> `separate:`
+To increase the accuracy of quantitative DNA sequencing experiments, Unique Molecular Identifiers may be used. UMIs are short sequences used to uniquely tag each molecule in a sample library, enabling precise identification of read duplicates. They must be added during library preparation and prior to sequencing, therefore require appropriate arrangements with your sequencing provider.
+
+Most tools capable of taking UMIs into consideration during an analysis workflow, expect the respective UMI sequence to be embedded into the read's ID. Please consult your tools' manuals regarding the exact specification.
+
+For some some library preparation kits and sequencing adapters, the UMI sequence needs to be read together with the index from the antisense strand. Consequently, it will be output as a separate FastQ file during the demultiplexing process.
```

Comment:
```suggestion
For some library preparation kits and sequencing adapters, the UMI sequence needs to be read together with the index from the antisense strand. Consequently, it will be output as a separate FastQ file during the demultiplexing process.
```